join.sh

```diff
@@ -16,10 +23,16 @@ cat <<EOS > /etc/docker/daemon.json
 }
 EOS
 
+#Configuration for Docker Behind a Proxy
+if [ $USE_PROXY == true ]; then
+  mkdir -p /etc/systemd/system/docker.service.d
+  CreateProxyConfiguraton
+fi
+
 export DEBIAN_FRONTEND=noninteractive
 apt-get update -qq
 apt-get install -y -qq --no-install-recommends \
-    docker.io \
+    docker.io=$DOCKER_VERSION \
```

Comment:
What if we want to step outside the realm of docker.io, e.g. use packages from apt.docker.io?

The versions available via docker.io package I believe cap off at 1.12.  I can't say for sure at this point, but its possible whatever "reliable and robust" stack we come up, which would include a suggested docker version, with may step outside whats available via the Canonical docker.io package.  Perhaps a path to ask for an alternative docker source, e.g. apt.docker.io or whatever the common ones are?

Thoughts?